pymaker/__init__.py

```diff
@@ -593,14 +630,18 @@ async def transact_async(self, **kwargs) -> Optional[Receipt]:
 
             replaced_tx.replaced = True
             self.nonce = replaced_tx.nonce
-
-        # Initialize variables which will be used in the main loop.
-        tx_hashes = []
-        initial_time = time.time()
-        gas_price_last = 0
+            # Gas should be calculated from the original time of submission
+            self.initial_time = replaced_tx.initial_time if replaced_tx.initial_time else time.time()
+            # Use gas strategy from the original transaction if one was not provided
+            if 'gas_price' not in kwargs:
+                self.gas_price = replaced_tx.gas_price if replaced_tx.gas_price else DefaultGasPrice()
+            self.gas_price_last = replaced_tx.gas_price_last
+            # Detain replacement until gas strategy produces a price acceptable to the node
+            if replaced_tx.tx_hashes:
+                self.tx_hashes = [replaced_tx.tx_hashes[-1]]
```

Comment:
I would extract `replaced_tx.tx_hashes[-1]` as a separate variable `tx_with_acceptable_gas_price` for the readability sake.

Reply:
Will do; I shall call it `most_recent_tx`.